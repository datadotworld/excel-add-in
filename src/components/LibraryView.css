/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
.items-view .item-selector .section-header {
  display: flex;
  justify-content: space-between;
}

.items-view .item-selector .section-header .add-icon {
  width: 15px;
  height: 15px;
  margin-right: 5px;
  cursor: pointer;
}

.items-view .item-selector .section-header .add-icon .stroke {
  stroke: #4376b2;
}

.items-view .section-header .title {
  font-size: 18px;
  font-weight: 700;
}

.items-view .loaderoverlay {
  position: inherit;
  top: initial;
  bottom: initial;
  left: initial;
  right: initial;
}

.items-view .item-selector .section-header button {
  border: none;
  box-shadow: none;
  padding: 0;
  color: #4376b2;
  float: right;
  font-size: 15px;
  font-weight: bold;
  margin-top: -8px;
  margin-bottom: 4px;
}

.message {
  padding: 20px;
}
