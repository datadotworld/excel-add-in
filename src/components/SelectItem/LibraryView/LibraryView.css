/*
 * Copyright 2018 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */

.library-view-container {
  position: absolute;
  top: 50px;
  left: 0px;
  width: 100%;
  height: 100%;
  z-index: 100;
  background-color: white;
}

.library-view-container .item-selector .section-header {
  display: flex;
  justify-content: space-between;
}

.library-view-container .item-selector .section-header .add-icon {
  width: 15px;
  height: 15px;
  margin-right: 5px;
  cursor: pointer;
}

.library-view-container .item-selector .section-header .add-icon .stroke {
  stroke: #4376b2;
}

.library-view-container .section-header .title {
  font-size: 18px;
  font-weight: 700;
}

.library-view-container .loaderoverlay {
  position: inherit;
  top: initial;
  bottom: initial;
  left: initial;
  right: initial;
}

.library-view-container .item-selector .section-header button {
  border: none;
  box-shadow: none;
  padding: 0;
  color: #4376b2;
  float: right;
  font-size: 15px;
  font-weight: bold;
  margin-top: -8px;
  margin-bottom: 4px;
}

.message {
  padding: 20px;
}
