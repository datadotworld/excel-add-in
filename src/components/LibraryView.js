/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { Button, Grid, Row, ControlLabel } from 'react-bootstrap';
import './LibraryView.css';
import LibraryItem from './LibraryItem';
import LoadingAnimation from './LoadingAnimation';
import { sortByOwnerAndTitle } from '../util';

class LibraryView extends Component {
  static propTypes = {
    onSelect: PropTypes.func,
    loading: PropTypes.bool,
    toggleList: PropTypes.func,
    toggleShowForm: PropTypes.func,
    getItems: PropTypes.func
  };

  state = {
    items: []
  };

  componentDidMount() {
    const { onlyShowWritableDatasets } = this.props;
    this.props.getItems(onlyShowWritableDatasets).then((items) => {
      this.setState({ items });
    });
  }

  render() {
    const {
      loading,
      toggleShowForm,
      toggleList,
      onSelect,
      isProjects
    } = this.props;
    const { items } = this.state;

    const sortedItems = sortByOwnerAndTitle(items);
    const entries = sortedItems.map((item) => {
      return (
        <LibraryItem
          item={item}
          key={`${item.owner}/${item.id}`}
          buttonText="Link"
          buttonHandler={onSelect}
          isProject={isProjects || item.isProject}
        />
      );
    });

    return (
      <Grid className="items-view">
        <div className="item-selector">
          <Row className="center-block section-header">
            <ControlLabel>{`Select a ${
              isProjects ? 'project' : 'dataset or project'
            }`}</ControlLabel>
            <Button bsStyle="default" onClick={() => toggleList()}>
              Cancel
            </Button>
          </Row>
          {loading && <LoadingAnimation label="Fetching items..." />}
          {!!items.length && !loading && (
            <Row className="center-block">
              <div>
                <LibraryItem
                  buttonText="Link"
                  item={{ isCreate: true }}
                  buttonHandler={toggleShowForm}
                  isProject={isProjects}
                />
                {entries}
              </div>
            </Row>
          )}
          {!items.length && !loading && (
            <Row className="center-block no-datasets">
              <div className="message">
                {isProjects
                  ? "You haven't created any projects to upload the insight to."
                  : "You haven't created any datasets to link data to."}
              </div>
              <Button
                className="bottom-button"
                bsStyle="primary"
                onClick={toggleShowForm}
              >
                {isProjects ? 'Create a new project' : 'Create a new dataset'}
              </Button>
            </Row>
          )}
        </div>
      </Grid>
    );
  }
}

export default LibraryView;
