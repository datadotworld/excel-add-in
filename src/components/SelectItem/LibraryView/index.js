/*
 * Copyright 2018 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */

import React, { Component } from 'react';
import { Button, Grid, Row, ControlLabel } from 'react-bootstrap';
import LibraryItem from './LibraryItem';
import LoadingAnimation from '../../LoadingAnimation';

import './LibraryView.css';

export default class LibraryView extends Component {
  state = {
    items: [],
    loading: true
  };

  componentDidMount() {
    this.props.getItems().then((items) => {
      this.setState({ items, loading: false });
    });
  }

  sortItems = () => {
    const sortedItems = this.state.items.slice();

    // Sort by updated in descending order
    sortedItems.sort((a, b) => {
      const dateA = new Date(a.updated);
      const dateB = new Date(b.updated);

      if (dateA < dateB) {
        return 1;
      }
      if (dateA > dateB) {
        return -1;
      }

      return 0;
    });

    return sortedItems;
  };

  render() {
    const {
      isProjects,
      toggleShowForm,
      onSelect,
      close,
      hideCreateNew
    } = this.props;
    const { loading, items } = this.state;

    const sortedItems = this.sortItems(items);
    const entries = sortedItems.map((item) => {
      return (
        <LibraryItem
          item={item}
          key={`${item.owner}/${item.id}`}
          buttonText="Link"
          buttonHandler={onSelect}
          isProject={isProjects || item.isProject}
        />
      );
    });
    return (
      <Grid className="library-view-container">
        <div className="item-selector">
          <Row className="center-block section-header">
            <ControlLabel>{`Select a ${
              isProjects ? 'project' : 'dataset or project'
            }`}</ControlLabel>
            <Button bsStyle="default" onClick={() => close()}>
              Cancel
            </Button>
          </Row>
          {loading && <LoadingAnimation label="Fetching items..." />}
          {!!items.length && !loading && (
            <Row className="center-block">
              <div>
                {!hideCreateNew && (
                  <LibraryItem
                    buttonText="Link"
                    item={{ isCreate: true }}
                    buttonHandler={toggleShowForm}
                    isProject={isProjects}
                    close={close}
                  />
                )}
                {entries}
              </div>
            </Row>
          )}
          {!items.length && !loading && (
            <Row className="center-block no-datasets">
              <div className="message">
                {isProjects
                  ? "You haven't created any projects to upload the insight to."
                  : "You haven't created any datasets to link data to."}
              </div>
              <Button
                className="bottom-button"
                bsStyle="primary"
                onClick={toggleShowForm}
              >
                {isProjects ? 'Create a new project' : 'Create a new dataset'}
              </Button>
            </Row>
          )}
        </div>
      </Grid>
    );
  }
}
