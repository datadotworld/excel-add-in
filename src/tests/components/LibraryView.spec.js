/*
 * Copyright 2017 data.world, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This product includes software developed at
 * data.world, Inc. (http://data.world/).
 */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';

import LibraryView from '../../components/LibraryView';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <LibraryView
      getItems={() =>
        new Promise((resolve) => {
          resolve([]);
        })
      }
    />,
    div
  );
});

it('renders datasets view - loading state', () => {
  expect(
    renderer
      .create(
        <LibraryView
          loadingLibraries
          getItems={() =>
            new Promise((resolve) => {
              resolve([]);
            })
          }
        />
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders datasets view - no datasets', () => {
  expect(
    renderer
      .create(
        <LibraryView
          loading={false}
          getItems={() =>
            new Promise((resolve) => {
              resolve([]);
            })
          }
        />
      )
      .toJSON()
  ).toMatchSnapshot();
});

it('renders datasets view - datasets', () => {
  expect(
    renderer
      .create(
        <IntlProvider locale="en">
          <LibraryView
            loading={false}
            getItems={() =>
              new Promise((resolve) => {
                resolve([]);
              })
            }
          />
        </IntlProvider>
      )
      .toJSON()
  ).toMatchSnapshot();
});
