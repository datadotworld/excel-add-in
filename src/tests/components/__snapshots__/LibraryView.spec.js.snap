// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders datasets view - datasets 1`] = `
<div
  className="items-view container"
>
  <div
    className="item-selector"
  >
    <div
      className="center-block section-header row"
    >
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Select a dataset or project
      </label>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
    <div
      className="center-block no-datasets row"
    >
      <div
        className="message"
      >
        You haven't created any datasets to link data to.
      </div>
      <button
        className="bottom-button btn btn-primary"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        Create a new dataset
      </button>
    </div>
  </div>
</div>
`;

exports[`renders datasets view - loading state 1`] = `
<div
  className="items-view container"
>
  <div
    className="item-selector"
  >
    <div
      className="center-block section-header row"
    >
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Select a dataset or project
      </label>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
    <div
      className="center-block no-datasets row"
    >
      <div
        className="message"
      >
        You haven't created any datasets to link data to.
      </div>
      <button
        className="bottom-button btn btn-primary"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        Create a new dataset
      </button>
    </div>
  </div>
</div>
`;

exports[`renders datasets view - no datasets 1`] = `
<div
  className="items-view container"
>
  <div
    className="item-selector"
  >
    <div
      className="center-block section-header row"
    >
      <label
        className="control-label"
        htmlFor={undefined}
      >
        Select a dataset or project
      </label>
      <button
        className="btn btn-default"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Cancel
      </button>
    </div>
    <div
      className="center-block no-datasets row"
    >
      <div
        className="message"
      >
        You haven't created any datasets to link data to.
      </div>
      <button
        className="bottom-button btn btn-primary"
        disabled={false}
        onClick={undefined}
        type="button"
      >
        Create a new dataset
      </button>
    </div>
  </div>
</div>
`;
